python/ray/data/_internal/lazy_block_list.py

```diff
@@ -303,13 +303,15 @@ def _get_blocks_with_metadata(
             block_refs.append(block_ref)
             meta_refs.append(meta_ref)
         if context.block_splitting_enabled:
-            # If block splitting is enabled, fetch the partitions.
-            parts = ray.get(block_refs)
+            # If block splitting is enabled, fetch the partitions through generator.
+            read_progress_bar = ProgressBar("Read progress", total=len(block_refs))
```

Comment:
We might need to handle duplicates due to unioning the same dataset, as we did for the non-dynamic block splitting path below. This can happen in a follow-up PR, e.g. when we turn dynamic block splitting on by default.